### PR TITLE
Replace compiled Less colors with CSS var() function calls and generate light mode calls

### PR DESCRIPTION
**Pull request by lippserd**

### CODE REVIEW

**Comment by nilmerg** [comment]
By the way, this must not run for IE users. There must be a way to disable the automatism globally.

**Comment by lippserd** [comment]
@nilmerg Ready to review!

File: library/Icinga/Less/ColorProp.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    {
+        return $this->index;
+    }
+
+    /**
+     * @param int $index
+     *
+     * @return $this
+     */
+    public function setIndex(int $index)
```
Comment (nilmerg) [inline library/Icinga/Less/ColorProp.php:59]: Not compatible with PHP 5.6

File: library/Icinga/Less/ColorProp.php [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    {
+        return $this->name;
+    }
+
+    /**
+     * @param string $name
+     *
+     * @return $this
+     */
+    public function setName(string $name)
```
Comment (nilmerg) [inline library/Icinga/Less/ColorProp.php:79]: Not compatible with PHP 5.6

File: library/Icinga/Less/LightModeDefinition.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    {
+        return $this->name;
+    }
+
+    /**
+     * @param string $name
+     *
+     * @return $this
+     */
+    public function setName(string $name)
```
Comment (nilmerg) [inline library/Icinga/Less/LightModeDefinition.php:43]: Not compatible with PHP 5.6

File: library/Icinga/Less/ColorProp.php [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    /** @var Less_Tree_Color Color with which we created the ColorProp */
+    protected $color;
+
+    /** @var int */
+    protected $index;
+
+    /** @var string Color variable name */
+    protected $name;
+
+    private function __construct()
```
Comment (nilmerg) [inline library/Icinga/Less/ColorProp.php:25]: Not compatible with PHP 5.6 (access level change)

**Review by nilmerg — CHANGES_REQUESTED** [review]
Every file is missing a license header. I thought we do this only for ipl content? Here every file has one, why should these differ?

Please fully ensure this doesn't run for IE like it's being done for Javascript [here](https://github.com/Icinga/icingaweb2/blob/master/library/Icinga/Application/webrouter.php#L71-L72).

File: library/Icinga/Application/webrouter.php [hunk 5]
```diff
@@ -56,10 +56,12 @@
 
     switch ($path) {
         case 'css/icinga.css':
-            Stylesheet::send();
+            $forIe11 = (bool) preg_match('/Trident\/7.0;.*rv:11/', $_SERVER['HTTP_USER_AGENT']);
+            Stylesheet::send(false, $forIe11);
```
Comment (nilmerg) [inline library/Icinga/Application/webrouter.php:60]: `$forIe11` should ne negated, the one below as well

**Review by nilmerg — CHANGES_REQUESTED** [review]

**Review by nilmerg — APPROVED** [review]

**Review by nilmerg — APPROVED** [review]

**Pull request merged**